### PR TITLE
Try to figure out why zipalign fails [DONT MERGE]

### PR DESCRIPTION
**Pull request by dellis1972**

### CODE REVIEW

File: src/Xamarin.Android.Build.Tasks/Utilities/ZipArchiveEx.cs [hunk 1]
```diff
@@ -33,8 +34,10 @@ public void Flush ()
 				zip.Close ();
 				zip.Dispose ();
 				zip = null;
+				//File.Copy (archive, $"{flush}_{archive}", overwrite: true);
```
Comment (jonpryor) [inline src/Xamarin.Android.Build.Tasks/Utilities/ZipArchiveEx.cs]: Why is this line commented out?

Comment (dellis1972) [reply src/Xamarin.Android.Build.Tasks/Utilities/ZipArchiveEx.cs]: I need to re-write it as it broke the build. It was just really late last night when I saw the errors so it was quicker to just comment out for now.

**Comment by jonpryor** [comment]
Current thoughts (for historical purposes): For [this build](https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=3134301&view=ms.vss-test-web.build-test-results-tab), in **Artifacts** > **mac-msbuild-test-results** > **xa-test-results-10.1.99.29_be27206-Darwin-Release.zip**, the `Flush()` method is invoked 3 times:

```
 ls -lh obj/Debug/android/bin/UnnamedProject.UnnamedProject.*
-rw-r--r--@ 1 jon  staff   4.1M Oct 11 13:46 obj/Debug/android/bin/UnnamedProject.UnnamedProject.1.apknew
-rw-r--r--@ 1 jon  staff   4.1M Oct 11 13:46 obj/Debug/android/bin/UnnamedProject.UnnamedProject.2.apknew
-rw-r--r--@ 1 jon  staff   4.1M Oct 11 13:46 obj/Debug/android/bin/UnnamedProject.UnnamedProject.3.apknew
-rw-r--r--@ 1 jon  staff   5.4M Oct 11 13:46 obj/Debug/android/bin/UnnamedProject.UnnamedProject.apk
```

I had been under the impression that the initial `NOTICE` addition was the problem, because it's always `NOTICE` which is invalid.  However, `UnnamedProject.UnnamedProject.3.apknew` is both valid *and* contains `NOTICE`!

```bash
$ unzip -t obj/Debug/android/bin/UnnamedProject.UnnamedProject.3.apknew
...
    testing: NOTICE                   OK
```

What happens between `UnnamedProject.UnnamedProject.3.apknew` and `UnnamedProject.UnnamedProject.apk`?

```bash
$ unzip -l obj/Debug/android/bin/UnnamedProject.UnnamedProject.3.apknew > good.txt
$ unzip -l obj/Debug/android/bin/UnnamedProject.UnnamedProject.apk > bad.txt
$ diff -u good.txt bad.txt 
--- good.txt	2019-10-11 15:40:24.000000000 -0400
+++ bad.txt	2019-10-11 15:40:38.000000000 -0400
@@ -1,4 +1,4 @@
-Archive:  obj/Debug/android/bin/UnnamedProject.UnnamedProject.3.apknew
+Archive:  obj/Debug/android/bin/UnnamedProject.UnnamedProject.apk
   Length      Date    Time    Name
 ---------  ---------- -----   ----
      3440  01-01-1980 00:00   AndroidManifest.xml
@@ -27,5 +27,13 @@
      6184  10-11-2019 13:20   lib/x86_64/libxamarin-app.so
   1697576  10-11-2019 13:20   lib/x86_64/libmono-btls-shared.so
    256176  10-11-2019 13:20   lib/x86_64/libmono-profiler-log.so
+   680472  10-11-2019 13:20   lib/arm64-v8a/libmono-native.so
+    27704  10-11-2019 13:20   lib/arm64-v8a/libxamarin-debug-app-helper.so
+   714476  10-11-2019 13:20   lib/armeabi-v7a/libmono-native.so
+    41552  10-11-2019 13:20   lib/armeabi-v7a/libxamarin-debug-app-helper.so
+   788196  10-11-2019 13:21   lib/x86/libmono-native.so
+   817512  10-11-2019 13:21   lib/x86_64/libmono-native.so
+    23460  10-11-2019 13:21   lib/x86/libxamarin-debug-app-helper.so
+    29080  10-11-2019 13:21   lib/x86_64/libxamarin-debug-app-helper.so
 ---------                     -------
- 10971274                     26 files
+ 14093726                     34 files
```

Thus, `NOTICE` is corrupted *after* `libmono-native.so` and `libxamarin-debug-app-helper.so` are added for the supported ABIs.  It doesn't appear to be ABI-specific, in that `libxamarin-app.so` and other native libs already exist for all of the supported ABIs, so things only start failing with either/both `libmono-native.so` and `libxamarin-debug-app-helper.so`.

Thus, it doesn't appear to be due to [`BuildApk.AddRuntimeLibraries()`](https://github.com/xamarin/xamarin-android/blob/09191bbf069a2fc6be4b330daba16f923ea7a6dd/src/Xamarin.Android.Build.Tasks/Tasks/BuildApk.cs#L421-L441) -- which adds `libmono-profiler-log.so`, which is present & valid in `.3.apknew` -- but instead [`BuildApk.AddNativeLibraries()`](https://github.com/xamarin/xamarin-android/blob/09191bbf069a2fc6be4b330daba16f923ea7a6dd/src/Xamarin.Android.Build.Tasks/Tasks/BuildApk.cs#L443-L449).

How do these methods differ?  For starters, `BuildApk.AddRuntimeLibraries()` takes `ZipArchiveEx` as the first parameter, and eventually does `apk.Archive.AddEntry(...)`.

`BuildApk.AddNativeLibraries()`, meanwhile, takes an `ArchiveFileList` -- a `List<System.Tuple<string, string>>` -- as the first parameter, and thus doesn't appear to actually modify the `.apk` itself.  Actual adding to the `.apk` happens separately in `BuildApk.ExecuteWithAbi()` using `apk.Archive.AddFile()`: https://github.com/xamarin/xamarin-android/blob/09191bbf069a2fc6be4b330daba16f923ea7a6dd/src/Xamarin.Android.Build.Tasks/Tasks/BuildApk.cs#L147-L160 

Could the issue be `apk.Archive.AddEntry()` vs. `apk.Archive.AddFile()`?  But `classes.dex` is added via `AddFile()`, so that seems unlikely.

**Pull request closed without merging**